scripts/mvf_rechunk.py

```diff
@@ -48,19 +63,15 @@ def __init__(self, stream_name, array_name, store, chunk_info):
         shape = chunk_info['shape']
         chunks = chunk_info['chunks']
         dtype = chunk_info['dtype']
-        self.data = store.get_dask_array(full_name, chunks, dtype)
-        self.has_data = store.has_array(full_name, chunks, dtype)
-        self.lost_flags = da.map_blocks(
-            self._make_lost, chunks=chunks, dtype=np.uint8,
-            name='lost-flags-{}-{}'.format(self.stream_name, self.array_name))
-
-    def _make_lost(self, block_info):
-        loc = block_info[None]['array-location']
-        shape = block_info[None]['chunk-shape']
-        if self.has_data[block_info[None]['chunk-location']]:
-            return np.zeros(shape, np.uint8)
+        raw_data = store.get_dask_array(full_name, chunks, dtype, errors='none')
+        # raw_data has `None` objects instead of ndarrays for chunks with
+        # missing data. That's not actually valid as a dask array, but we use
+        # it to produce lost flags (similarly to datasources.py).
+        self.lost_flags = da.map_blocks(_make_lost, raw_data, dtype=np.uint8)
+        if array_name != 'flags':
+            self.data = da.map_blocks(_fill_missing, raw_data, fill_value, dtype=raw_data.dtype)
```

Comment:
I don't see a definition for `fill_value`...

Reply:
Urgh, I've probably made a mess of it yet again. Let me actually test it in the morning and get back to you.

Reply:
Ok, should be fixed now, and this time I actually tested it.